src/detect-parse.c

```diff
@@ -973,41 +916,137 @@ static int SigParseAction(Signature *s, const char *action)
     }
 }
 
+/**
+ * \brief Parse the next token in rule.
+ *
+ * For rule parsing a token is considered to be a string of characters
+ * separated by white space.
+ *
+ * \param input double pointer to input buffer, will be advanced as input is
+ *     parsed.
+ * \param output buffer to copy token into.
+ * \param output_size length of output buffer.
+ */
+static inline int SigParseToken(const char **input, char *output,
+    const size_t output_size)
+{
+    size_t len = *input == NULL ? 0 : strlen(*input);
+    size_t token_len = 0;
+    const char *token = *input;
+
+    if (!len) {
+        return 0;
+    }
+
+    while (len && isblank(*token)) {
+        token++;
+        len--;
+    }
+
+    char *endptr = strpbrk(token, " \t\n\r");
+    if (endptr != NULL) {
+        token_len = endptr - token;
+    } else {
+        token_len = len;
+    }
+    strlcpy(output, token, token_len + 1);
```

Comment:
this is invalid use of strlcpy, you're supposed to pass the output buffer size as length

Reply:
Perhaps I'll replace the strdup with a copy on the stack, so the NULs can be put back in and strlcpy can be used as intended.